### PR TITLE
Scx1332/deposits

### PR DESCRIPTION
**Pull request by scx1332**

### CODE REVIEW

File: specs/deposits.md
Comment (nieznanysprawiciel) [inline specs/deposits.md:30]: It would be good to have generalized description here, before we show contract details. And I would move flow chart somewhere here. Documentation should rather take top-down approach

File: specs/deposits.md
Comment (nieznanysprawiciel) [inline specs/deposits.md:178]: I would move this up, this would allow reader to better understand general concept

File: specs/deposits.md
Comment (nieznanysprawiciel) [inline specs/deposits.md:245]: Plus we don't have any part about deposit parameters validation by service

File: specs/deposits.md
Comment (nieznanysprawiciel) [inline specs/deposits.md:78]: We should explain that these events serve a purpose of replacing lack of list deposits function on contract

File: specs/deposits.md
Comment (scx1332) [reply specs/deposits.md:62]: ok

**Review by nieznanysprawiciel — CHANGES_REQUESTED** [review]
I don't think you addressed comments from my previous review, only some of them. It is not enough to move things around.
If you read this spec, do you know what calls to yagna you have to make, to use this feature? There is Deposit struct showed, but it is not shown how to use it.
In contract description there is no separation of functions that are interface for yagna.

In general there is everything mentioned, but there are no connections made between things, so this document is not enough to be thought of as complete.

**Review by nieznanysprawiciel — APPROVED** [review]
Approving, but we are waiting for Rekuc approval as well

File: specs/deposits.md
Comment (golmek) [inline specs/deposits.md:380]: Either fill it up or remove completely before merge to master.

**Review by golmek — CHANGES_REQUESTED** [review]
@scx1332 please look at those before merge to master.

**Review by prekucki — APPROVED** [review]

**Review by golmek — APPROVED** [review]

**Pull request merged**
